### PR TITLE
Upgrade to MUI v6

### PR DESCRIPTION
**Pull request by jamesricky**
## Changeset

-   [x] I have verified if my change requires a changeset

## Related tasks and documents

Task: https://vivid-planet.atlassian.net/browse/COM-1123
V8 Upgrade: https://github.com/vivid-planet/comet-upgrade/pull/23

## Checklist

- [x] Check if breaking changes affect usages of our components and fix if necessary
   - [x] [Accordion](https://mui.com/material-ui/migration/upgrade-to-v6/#accordion)
   - [x] [Autocomplete](https://mui.com/material-ui/migration/upgrade-to-v6/#autocomplete)
   - [x] [Chip](https://mui.com/material-ui/migration/upgrade-to-v6/#chip)
   - [x] [Divider](https://mui.com/material-ui/migration/upgrade-to-v6/#divider)
   - [x] [ListItem](https://mui.com/material-ui/migration/upgrade-to-v6/#listitem)
   - [x] [Loading Button](https://mui.com/material-ui/migration/upgrade-to-v6/#loading-button)
   - [x] [Typography](https://mui.com/material-ui/migration/upgrade-to-v6/#typography)
- [x] Check if we still need `@mui/lab`
   - Still required for usage of MUIs `LoadingButton` in our `FeedbackButton` component
- [x] Run relevant codemods

### CODE REVIEW

File: .changeset/perfect-mangos-greet.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+"@comet/admin-date-time": major
+"@comet/blocks-admin": major
+"@comet/admin-icons": major
+"@comet/admin-theme": major
+"@comet/admin-rte": major
+"@comet/cms-admin": major
+"@comet/admin": major
+---
+
+Migrate to MUI v6
```
Comment (johnnyomair) [inline .changeset/perfect-mangos-greet.md:13]: Could you please start the upgrade guide as well?

Comment (jamesricky) [reply .changeset/perfect-mangos-greet.md:13]: Do you mean this in the docs? 
https://docs.comet-dxp.com/docs/migration/migration-from-v6-to-v7

File: packages/admin/cms-admin/src/dam/DataGrid/thumbnail/DamThumbnail.tsx [hunk 2]
```diff
@@ -66,15 +66,7 @@ export const DamThumbnail = ({ asset }: DamThumbnailProps) => {
             thumbnail = (
                 <>
                     <ImageThumbnail onMouseOver={handleMouseOver} onMouseLeave={handleMouseLeave} src={asset.image.thumbnailUrl} />
-                    <Popper
-                        sx={{ zIndex: 1301 }}
-                        open={open}
-                        anchorEl={anchorEl}
-                        placement="auto-end"
-                        onResize={undefined}
-                        onResizeCapture={undefined}
```
Comment (thomasdax98) [inline packages/admin/cms-admin/src/dam/DataGrid/thumbnail/DamThumbnail.tsx:75]: Does this have anything to do with the update or was it unnecessary before?

(since I can't find any info relating to this change in the changelog or migration guide)

File: packages/admin/admin/src/dataGrid/CrudMoreActionsMenu.tsx [hunk 3]
```diff
@@ -52,7 +52,7 @@ interface CrudMoreActionsGroupProps {
 function CrudMoreActionsGroup({ groupTitle, children, menuListProps, typographyProps }: PropsWithChildren<CrudMoreActionsGroupProps>) {
     return (
         <>
-            <Typography variant="overline" color={(theme) => theme.palette.grey[500]} sx={{ padding: "20px 15px 0 15px" }} {...typographyProps}>
+            <Typography variant="overline" color="grey.500" sx={{ padding: "20px 15px 0 15px" }} {...typographyProps}>
```
Comment (thomasdax98) [inline packages/admin/admin/src/dataGrid/CrudMoreActionsMenu.tsx:55]: Is color typesafe so it has autocompletion?

File: .changeset/perfect-mangos-greet.md (hunk 1 above)
Comment (jamesricky) [reply .changeset/perfect-mangos-greet.md:13]: Do I need to add something in https://github.com/vivid-planet/comet-upgrade as well then?

**Review by thomasdax98 — APPROVED** [review]

File: .changeset/perfect-mangos-greet.md (hunk 1 above)
Comment (thomasdax98) [reply .changeset/perfect-mangos-greet.md:13]: > Do you mean this in the docs? (just for v7-to-v8)

Yes

> Do I need to add something in https://github.com/vivid-planet/comet-upgrade as well then?

Theoretically we could execute the MUI codemods in our upgrade script

File: packages/admin/cms-admin/src/dam/DataGrid/thumbnail/DamThumbnail.tsx (hunk 2 above)
Comment (jamesricky) [reply packages/admin/cms-admin/src/dam/DataGrid/thumbnail/DamThumbnail.tsx:75]: It was only necessary due to a typescript error that has been fixed with MUI v6

File: packages/admin/admin/src/dataGrid/CrudMoreActionsMenu.tsx (hunk 3 above)
Comment (jamesricky) [reply packages/admin/admin/src/dataGrid/CrudMoreActionsMenu.tsx:55]: No, this can be any `string` e.g. to support css color values

File: .changeset/perfect-mangos-greet.md (hunk 1 above)
Comment (jamesricky) [reply .changeset/perfect-mangos-greet.md:13]: Added to the docs in b9bc8bc05fbeea91305f8951dcb87167a8a0b695
And codemods are executed in https://github.com/vivid-planet/comet-upgrade/pull/23

**Review by thomasdax98 — APPROVED** [review]

File: .changeset/perfect-mangos-greet.md (hunk 1 above)
Comment (johnnyomair) [inline .changeset/perfect-mangos-greet.md:13]:
```suggestion
Upgrade to MUI v6
```

I'd call this upgrade, since we doesn't really migrate from one thing to another

File: packages/admin/admin/src/common/Tooltip.tsx [hunk 4]
```diff
@@ -236,7 +239,7 @@ export const Tooltip = (inProps: TooltipProps) => {
                 disableTouchListener
                 {...commonTooltipProps}
             >
-                {cloneElement(children, { onClick: toggleTooltip })}
+                {cloneElement(children as ReactElement<any, any>, { onClick: toggleTooltip })}
```
Comment (johnnyomair) [inline packages/admin/admin/src/common/Tooltip.tsx:242]: Why is this necessary?

File: packages/admin/blocks-admin/src/iframebridge/HoverPreviewComponent.tsx [hunk 5]
```diff
@@ -38,13 +38,13 @@ export const HoverPreviewComponent = ({ children, componentSlug }: PropsWithChil
     return (
         <sc.Root
             ref={rootEl}
-            onMouseOver={(e) => {
+            onMouseOver={(e: React.MouseEvent<HTMLDivElement>) => {
```
Comment (johnnyomair) [inline packages/admin/blocks-admin/src/iframebridge/HoverPreviewComponent.tsx:41]: Why is this explicit type annotation necessary? Wouldn't it be better to fix the type of `Root`?

File: packages/admin/admin/src/common/Tooltip.tsx (hunk 4 above)
Comment (jamesricky) [reply packages/admin/admin/src/common/Tooltip.tsx:242]: The type of `children` in MUIs `Tooltip` was changed from `ReactElement<any, any>` to `React.ReactElement<unknown, any>`. 

Since this change typescript no longer ignores that `onClick` might not be available on whatever element `children` is, so it is not allowed to set `onClick`. 

I wasn't exactly sure how to check if the element is supported and since this doesn't change anything in the behavior it didn't seem worth investing a lot of time into fixing this 😅

File: packages/admin/blocks-admin/src/iframebridge/HoverPreviewComponent.tsx (hunk 5 above)
Comment (jamesricky) [reply packages/admin/blocks-admin/src/iframebridge/HoverPreviewComponent.tsx]: Fixed in 5487afda5dea233d5f53fbe951888c082da877ce

I don't really understand why but changing `styled("div")` to `styled(Box)` fixes the type. Maybe not an optimal solution but since this has no effect on styling an it's the only place where it's really an issue it's probably fine.

**Review by johnnyomair — APPROVED** [review]
